airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -293,6 +318,61 @@ private void reportFailure(final ConnectionUpdaterInput connectionUpdaterInput,
     }
   }
 
+  private SyncCheckConnectionFailure checkConnections(GenerateInputActivity.GeneratedJobInput jobInputs) {
+    final JobRunConfig jobRunConfig = jobInputs.getJobRunConfig();
+    final StandardSyncInput syncInput = jobInputs.getSyncInput();
+    final JsonNode sourceConfig = syncInput.getSourceConfiguration();
+    final JsonNode destinationConfig = syncInput.getDestinationConfiguration();
+    final IntegrationLauncherConfig sourceLauncherConfig = jobInputs.getSourceLauncherConfig();
+    final IntegrationLauncherConfig destinationLauncherConfig = jobInputs.getDestinationLauncherConfig();
+    final Long jobId = jobRunConfig.getJobId() != null ? Long.valueOf(jobRunConfig.getJobId()) : 1L;
+    final Integer attemptId = jobRunConfig.getAttemptId() != null ? Math.toIntExact(jobRunConfig.getAttemptId()) : 1;
+    SyncCheckConnectionFailure checkFailure = new SyncCheckConnectionFailure(jobId, attemptId);
+
+    final int attemptCreationVersion =
+        Workflow.getVersion(CHECK_BEFORE_SYNC_TAG, Workflow.DEFAULT_VERSION, CHECK_BEFORE_SYNC_CURRENT_VERSION);
+
+    if (attemptCreationVersion >= CHECK_BEFORE_SYNC_CURRENT_VERSION) {
+      final StandardCheckConnectionInput sourceConfiguration = new StandardCheckConnectionInput().withConnectionConfiguration(sourceConfig);
+      final CheckConnectionInput checkSourceInput = new CheckConnectionInput(jobRunConfig, sourceLauncherConfig, sourceConfiguration);
+      final String launchSourceDockerImage = sourceLauncherConfig.getDockerImage();
+
+      if (WorkerConstants.RESET_JOB_SOURCE_DOCKER_IMAGE_STUB.equals(launchSourceDockerImage) || checkFailure.isFailed()) {
```

Comment:
It feels slightly brittle to be relying on the docker image name to decide whether this is a reset connection job or not. Could we instead just pass the `workflowState.isResetConnection()` boolean value into this `checkConnections()` method and use that to decide this behavior?

Reply:
Nice! I didn't know `isResetConnection()` was available!